### PR TITLE
Better error message when only a symbol is passed to constraint macro

### LINKED ISSUE #728: uninformative error message when not passing an expr into a constraint
so i hadn't used SDP constraint before and typed `@addSDPConstraint(mod, A)` and got an error:
> ERROR: LoadError: type Symbol has no field head
>  in include at /Applications/Julia-0.4.3.app/Contents/Resources/julia/lib/julia/sys.dylib
>  in include_from_node1 at /Applications/Julia-0.4.3.app/Contents/Resources/julia/lib/julia/sys.dylib
>  in process_options at /Applications/Julia-0.4.3.app/Contents/Resources/julia/lib/julia/sys.dylib
>  in _start at /Applications/Julia-0.4.3.app/Contents/Resources/julia/lib/julia/sys.dylib
> while loading blah blah

which didn't help so of course I called Miles and he said you're dumb, use `@addSDPConstraint(mod, A >= 0)`, at which point I followed his advice and things worked i.e. different errors got thrown instead yay

### PR DESCRIPTION
**Pull request by joehuchette**
closes #728

### CODE REVIEW

**Comment by mlubin** [comment]
Could we merge after #726 to avoid conflicts?

File: src/macros.jl [hunk 1]
```diff
@@ -298,6 +298,9 @@ macro constraint(args...)
     end
     m = args[1]
     x = args[2]
+    if isa(x, Symbol)
+        error("in @addConstraint($(join(args,','))): Incomplete constraint specification $x. Are you missing a comparison (<=, >=, or ==)?")
```
Comment (mlubin) [inline src/macros.jl]: Macro names need an update

**Comment by mlubin** [comment]
Add a test so this doesn't tick down our coverage?

**Comment by mlubin** [comment]
Does ``@fact_throws`` actually work for compile-time errors? not sure about that

**Comment by joehuchette** [comment]
Nope. How to test?

**Comment by mlubin** [comment]
I think you can do ``@fact_throws`` on ``macroexpand``.

**Pull request merged**